huaweicloud/services/ces/resource_huaweicloud_ces_alarmrule.go

```diff
@@ -303,53 +364,133 @@ func buildAlarmAction(d *schema.ResourceData, name string) []alarmrule.ActionOpt
 	return nil
 }
 
-func buildAlarmCondition(d *schema.ResourceData) alarmrule.ConditionOpts {
-	var opts alarmrule.ConditionOpts
+func buildDimensionsOpts(dimensionsRaw []interface{}) [][]alarmrulev2.DimensionOpts {
+	if len(dimensionsRaw) < 1 {
+		return [][]alarmrulev2.DimensionOpts{}
+	}
+	resources := make([][]alarmrulev2.DimensionOpts, len(dimensionsRaw))
+	for i, dimensionRaw := range dimensionsRaw {
+		dimension := dimensionRaw.(map[string]interface{})
+		resources[i] = []alarmrulev2.DimensionOpts{
+			{
+				Name:  dimension["name"].(string),
+				Value: dimension["value"].(string),
+			},
+		}
+	}
 
-	rawCondition := d.Get("condition").([]interface{})
-	if len(rawCondition) == 1 {
-		condition := rawCondition[0].(map[string]interface{})
+	return resources
+}
 
-		opts.Period = condition["period"].(int)
-		opts.Filter = condition["filter"].(string)
-		opts.ComparisonOperator = condition["comparison_operator"].(string)
-		opts.Value = condition["value"].(float64)
-		opts.Unit = condition["unit"].(string)
-		opts.Count = condition["count"].(int)
-		opts.SuppressDuration = condition["suppress_duration"].(int)
+func buildResourcesOpts(d *schema.ResourceData) ([][]alarmrulev2.DimensionOpts, string, string) {
+	metricRaw := d.Get("metric").([]interface{})
+	if len(metricRaw) != 1 {
+		return nil, "", ""
 	}
 
-	return opts
+	metric := metricRaw[0].(map[string]interface{})
+	dimensionsRaw := metric["dimensions"].(*schema.Set).List()
+
+	return buildDimensionsOpts(dimensionsRaw), metric["namespace"].(string), metric["metric_name"].(string)
+}
+
+func buildNotificationsOpts(d *schema.ResourceData, name string) []alarmrulev2.NotificationOpts {
+	if v, ok := d.GetOk(name); ok {
+		notificationOptsRaw := v.([]interface{})
+		notificationOpts := make([]alarmrulev2.NotificationOpts, len(notificationOptsRaw))
+		for i, notificationOptRaw := range notificationOptsRaw {
+			notificationOpt := notificationOptRaw.(map[string]interface{})
+
+			notificationListRaw := notificationOpt["notification_list"].([]interface{})
+			notificationList := make([]string, len(notificationListRaw))
+			for j, notification := range notificationListRaw {
+				notificationList[j] = notification.(string)
+			}
+
+			notificationOpts[i] = alarmrulev2.NotificationOpts{
+				Type:             notificationOpt["type"].(string),
+				NotificationList: notificationList,
+			}
+		}
+		return notificationOpts
+	}
+	return nil
+}
+
+func buildPoliciesOpts(d *schema.ResourceData, metricName string) []alarmrulev2.PolicyOpts {
```

Comment:
```suggestion
func buildPoliciesOpts(d *schema.ResourceData, globalMetricName string) []alarmrulev2.PolicyOpts {
```

maybe `globalMetricName ` is easy to understand.

Reply:
fixed